### ISSUE TITLE
ValueError: high is out of bounds for int32

### ISSUE THREAD

**Issue by 321HG** [SimonBlanke/Gradient-Free-Optimizers#11]
How can I solve this problem, please?
...............

D:\...................\search.py in set_random_seed(nth_process, random_state)
     33 
     34     if random_state is None:
---> 35         random_state = np.random.randint(0, high=2 ** 32 - 2)
     36 
     37     random.seed(random_state + nth_process)

mtrand.pyx in numpy.random.mtrand.RandomState.randint()

_bounded_integers.pyx in numpy.random._bounded_integers._rand_int32()

ValueError: high is out of bounds for int32

**Comment by SimonBlanke** [comment]
Hello @321HG,

i need some additional information to find the problem. Please use the template for the "bug report":

**Describe the bug**

**Code to reproduce the behavior**

**Error message from command line**

**System information:**
 - OS Platform and Distribution
 - Python version
 - Gradient-Free-Optimizers version

**Additional context**

**Comment by architrathore** [comment]
I've also run into this problem in `set_random_seed`. A quick fix is to pass `dtype=np.int64` to method like so:

```python
 random_state = np.random.randint(0, high=2 ** 32 - 2, dtype=np.int64)
```
It might also make sense to use a lower range for setting the seed since numpy's int32, which is the default dtype that is used, goes from -2^31 to 2^31.

**Comment by 321HG** [comment]
> Hello @321HG,
> 
> i need some additional information to find the problem. Please use the template for the "bug report":
> 
> **Describe the bug**
> 
> **Code to reproduce the behavior**
> 
> **Error message from command line**
> 
> **System information:**
> 
> * OS Platform and Distribution
> * Python version
> * Gradient-Free-Optimizers version
> 
> **Additional context**

**Comment by 321HG** [comment]
> 我也曾在中遇到这个问题`set_random_seed`。一个快速的解决`dtype=np.int64`方法是传递给这样的方法：
> 
> ```python
>  random_state  =  np。随机的。randint（0，高= 2  **  32  -  2，D型细胞= NP。的int64）
> ```
> 
> 使用较小的范围来设置种子可能也很有意义，因为numpy的int32（即使用的默认dtype）从-2 ^ 31变为2 ^ 31。

Thank you very much.👍👍👍👍

**Comment by SimonBlanke** [comment]
@321HG and @architrathore what operating system are you using? I am on Ubuntu 20.04 and there were never any problems. If this problem is that common I should fix it quickly. Before that I need a way to reproduce it.

**Comment by architrathore** [comment]
I am on windows. Running the same code from Windows Subsystem for Linux (WSL2) works fine though.
This seems to be a problem with the Win64 platform, numpy defaults to int32 on windows while it defaults to int64 on Linux.

Relevant StackOverflow thread: https://stackoverflow.com/a/36279549

**Comment by SimonBlanke** [comment]
@architrathore thank you very much! In this problem: `2 ** 32 - 2 ` i confused the range of uint32 and int32.

**Comment by SimonBlanke** [comment]
I just released version 3.0.0. This version contains the fix for this issue (as well as several other improvements + fixes).

@architrathore Could you verify, that this issue is fixed in v3.0.0? I do not have access to windows.
